bigquery/google/cloud/bigquery/client.py

```diff
@@ -1527,7 +1537,7 @@ def load_table_from_dataframe(
                         PendingDeprecationWarning,
                         stacklevel=2,
                     )
-                dataframe.to_parquet(tmppath)
+                dataframe.to_parquet(tmppath, compression=parquet_compression)
```

Comment:
We do need to modify `_pandas_helpers.dataframe_to_parquet` as well. See: https://github.com/googleapis/google-cloud-python/blob/1b3e82270a8ca72cbfd90a6a9a73b641a57111bd/bigquery/google/cloud/bigquery/_pandas_helpers.py#L230 The underlying [pyarrow.parquet.write_table](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_table.html) function also takes a compression argument.

Long-term, I expect the `_pandas_helpers.dataframe_to_parquet` function to get used more often than the `dataframe.to_parquet` method. We'll want to start fetching the table schema if not provided and use that for pandas to BigQuery type conversions https://github.com/googleapis/google-cloud-python/issues/8142.

Reply:
There is a mismatch between the two methods, the pyarrow's accepts a richer range of compression methods.

Having either two different compression parameters, or a single parameter that accepts different values depending on the context, could be confusing to the end users, thus I will only allow the compression methods supported by both.

However, if there are uses cases that would need specific support for LZO, LZ4, and ZSTD as well, please let me know. Probably there aren't, because to date the compression method has not been exposed anyway?